### PR TITLE
Add explanation of water balance error

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/1820.

### CODE REVIEW

File: docs/concept/equations.qmd [hunk 1]
```diff
@@ -79,6 +79,63 @@ $$
 $$
 i.e. $J_{i,j}$ quantifies how $f_j$. the time derivative of state $j$, changes with respect to changes in state $i$. Most of these entries are $0$, because flows in distant parts of the model do not depend on each other.
 
+## The water balance error
+
+The water balance error quantifies how well the water volume in the model is conserved for each basin over an output save period, i.e. whether no water erroneously appears or disappears. It looks at the storage rate
```
Comment (visr) [inline docs/concept/equations.qmd:84]:
```suggestion
The water balance error quantifies how well the water volume in the model is conserved for each Basin over an output save period, i.e. whether no water erroneously appears or disappears. It looks at the storage rate
```

File: docs/concept/equations.qmd [hunk 2]
```diff
@@ -79,6 +79,63 @@ $$
...
+$$
+	\text{storage rate} = \frac{\Delta S_b}{\Delta t}
+$$
+
+in a basin over a time period $\Delta t = t_1 - t_0$ and compares that to the total inflows and outflows of that basin over that period. More precisely, we first compute the total inflow and outflow, where:
```
Comment (visr) [inline docs/concept/equations.qmd:89]:
```suggestion
in a Basin over a time period $\Delta t = t_1 - t_0$ and compares that to the total inflows and outflows of that Basin over that period. More precisely, we first compute the total inflow and outflow, where:
```

File: docs/concept/equations.qmd [hunk 3]
```diff
@@ -79,6 +79,63 @@ $$
...
+
+- $\text{total inflow}$: the precipitation, drainage and horizontal flows into the basin;
+- $\text{total outflow}$: the evaporation, infiltration and horizontal flows out of the basin.
```
Comment (visr) [inline docs/concept/equations.qmd:92]:
```suggestion
- $\text{total inflow}$: the precipitation, drainage and horizontal flows into the Basin;
- $\text{total outflow}$: the evaporation, infiltration and horizontal flows out of the Basin.
```

File: docs/concept/equations.qmd [hunk 4]
```diff
@@ -79,6 +79,63 @@ $$
...
+
+Whether a flow is an inflow or an outflow depends on whether the flow contributes to or takes from the basin storage, and thus this is independent of the edge direction. This is determined for each solver timestep individually.
```
Comment (visr) [inline docs/concept/equations.qmd:94]:
```suggestion
Whether a flow is an inflow or an outflow depends on whether the flow contributes to or takes from the Basin storage, and thus this is independent of the edge direction. This is determined for each solver timestep individually.
```

File: docs/concept/equations.qmd [hunk 5]
```diff
@@ -79,6 +79,63 @@ $$
...
+
+Then from this we compute the errors:
+
+$$
+	\begin{align}
+	\text{balance error} =&& \text{storage rate} - (\text{total inflow} - \text{total outflow}) \\
+	\text{relative error}=&& \frac{\text{absolute error}}{0.5(\text{total inflow} + \text{total outflow})}
+	\end{align}
+$$
+Hence the reference used for computing the relative error is the average of the total inflow and total outflow of the basin (which are both non-negative).
```
Comment (visr) [inline docs/concept/equations.qmd:104]:
```suggestion
Hence the reference used for computing the relative error is the average of the total inflow and total outflow of the Basin (which are both non-negative).
```

File: docs/concept/equations.qmd [hunk 6]
```diff
@@ -79,6 +79,63 @@ $$
...
+
+In extreme cases where storage rate is many orders of magnitude smaller than the storage itself, these computations can have floating point truncation errors which can lead to large relative errors.  This is however only when the storage is roughly $10^{15}$ or more times bigger than the storage rate.
+
+### Example calculation
+
+Say we have the following model:
+
+![](https://s3.deltares.nl/ribasim/doc-image/concept/equations/subnetwork.png){fig-align="left"}
+
+and we want to calculate the water balance error for basin 6. We have the following data:
```
Comment (visr) [inline docs/concept/equations.qmd:116]:
```suggestion
and we want to calculate the water balance error for Basin 6. We have the following data:
```

File: docs/reference/usage.qmd [hunk 7]
```diff
@@ -201,6 +201,8 @@ The Basin table contains:
   It can be used to check if the numerical error when solving the water balance is sufficiently small.
 - The `relative_error` is the fraction of the `balance_error` over the mean of the `total_inflow` and `total_outflow`.
 
+For a more in-depth explanation of the water balance error see [here](../concept/equations.qmd#the-water-balance-error).
```
Comment (visr) [inline docs/reference/usage.qmd:204]:
```suggestion
For a more in-depth explanation of the water balance error see [here](/concept/equations.qmd#the-water-balance-error).
```

File: docs/concept/equations.qmd [hunk 8]
```diff
@@ -79,6 +79,63 @@ $$
...
+
+And so we get
+
+$$
+\begin{align}
+	\text{storage rate} = && \frac{50.0 - 100.0}{10.0} &= & -6.0 \text{ m}^3/\text{s} \\
+	\text{total inflow} = && 5.0 + 3.5 &= & 8.5 \text{ m}^3/\text{s}\\
+	\text{total outflow} = && 10.0 + 4.0 &= & 14.0 \text{ m}^3/\text{s}\\
+	\text{balance error} = && -6.0 - (8.5 - 14.0) &= & -0.5 \text{ m}^3/\text{s}\\
+		\text{relative error} = && \frac{-0.5}{8.5 + 14.0} &\approx & -0.022
```
Comment (visr) [inline docs/concept/equations.qmd:134]:
```suggestion
	\text{relative error} = && \frac{-0.5}{8.5 + 14.0} &\approx & -0.022
```

**Review by visr — APPROVED** [review]

**Comment by SouthEndMusic** [comment]
Todo do:
- Add explanation of callback check
- In `usage.qmd`: add explanation of water balance tolerance options

**Pull request merged**